### ISSUE TITLE
Header matching fails due to terminating on comma

### ISSUE THREAD

**Issue by alexeits** [pact-foundation/pact-reference#353]
Given the following pact JSON:
```json
"headers": {
"Content-Type": "application/json",
"Set-Cookie": "auth=fake-token; HttpOnly"
},
"body": {},
"matchingRules": {
"$.headers.Content-Type": {
  "match": "regex",
  "regex": "^application\\/json(;|$)"
},
"$.headers.Set-Cookie": {
  "match": "regex",
  "regex": "^auth=[^;]+;.*\\s[Hh]ttp[Oo]nly(?:;|$)"
}
```
Complete JSON:  [agent-desktop-impersonation_tokens.json](https://github.com/pact-foundation/pact-reference/files/13811492/agent-desktop-impersonation_tokens.json)

and the provider responding with: 
```
auth=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE3MDQyMTczMDQsImlhdCI6MTcwNDIxMDEwNCwiYWN0aXZhdGVkQXQiOiIyMDI0LTAxLTAyVDE1OjQxOjQ1Ljg0NVoiLCJ1c2VySWQiOiJleGFtcGxlVXNlcklkIiwib3JnSWQiOiJleGFtcGxlT3JnSWQiLCJyb2xlSWRzIjpbImFnZW50Il0sInNhZ2FuVXNlciI6ImFkbWluQGV4YW1wbGUub3JnIiwic2Vzc2lvbiI6eyJpZCI6IkY2QklCd2xtUXNTckJvQ01CX0pEZXcifX0.kNXCzxDQjGEmEwEAoLC1JLLUYPTQjWgHheLlhEPlOc0; Path=/; Expires=Tue, 02 Jan 2024 17:41:44 GMT; Max-Age=7198; HttpOnly
```
which matches the `Set-Cookie` regex: https://regex101.com/r/bTjvJT/1,

the verification fails with:
```
1) Verifying a pact between agent-desktop and impersonation_tokens Given agent user exists - start impersonation
    1.1) includes header 'Set-Cookie' with value 'auth=fake-token; HttpOnly'
           Expected header 'Set-Cookie' to have value 'auth=fake-token; HttpOnly' but was 'auth=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE3MDQyMTczMDQsImlhdCI6MTcwNDIxMDEwNCwiYWN0aXZhdGVkQXQiOiIyMDI0LTAxLTAyVDE1OjQxOjQ1Ljg0NVoiLCJ1c2VySWQiOiJleGFtcGxlVXNlcklkIiwib3JnSWQiOiJleGFtcGxlT3JnSWQiLCJyb2xlSWRzIjpbImFnZW50Il0sInNhZ2FuVXNlciI6ImFkbWluQGV4YW1wbGUub3JnIiwic2Vzc2lvbiI6eyJpZCI6IkY2QklCd2xtUXNTckJvQ01CX0pEZXcifX0.kNXCzxDQjGEmEwEAoLC1JLLUYPTQjWgHheLlhEPlOc0; Path=/; Expires=Tue'
```

Notice that the match is terminated on the comma after "Tue".

**Comment by YOU54F** [comment]
This should be resolved in [libpact_ffi-v0.4.13](https://github.com/pact-foundation/pact-reference/releases/tag/libpact_ffi-v0.4.13) or higher.

Please let us know if this is still an issue

**Comment by alexeits** [comment]
@YOU54F: I confirm that it's resolved. We are currently on 0.4.16 and no longer see this issue.
